### PR TITLE
Add t3-env to AvaTax

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
## Scope of the PR

Adds [t3-env](https://github.com/t3-oss/t3-env) library for validating env variables.

I tested:
* Error flow (with Sentry capture)
* Working flow (calc taxes)
* e2e

## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/avatax/next.config.js
Comment (lkostrowski) [inline apps/avatax/next.config.js:1]: can we import t3 env here, so it validates during build?

Comment (krzysztofzuraw) [reply apps/avatax/next.config.js:1]: We can but we will have to install additional pkg: https://env.t3.gg/docs/nextjs#validate-schema-on-build-(recommended) as Next.js don't support importing ts files in config natively

File: apps/avatax/tsconfig.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "node",
+    "moduleResolution": "Bundler",
```
Comment (lkostrowski) [inline apps/avatax/tsconfig.json:12]: is it safe?

File: apps/avatax/next.config.js
Comment (lkostrowski) [reply apps/avatax/next.config.js:1]: ok, next 15 does so it can wait

File: apps/avatax/tsconfig.json (hunk 1 above)
Comment (krzysztofzuraw) [reply apps/avatax/tsconfig.json:12]: I checked and Next.js supports it from at least 1 year: https://github.com/vercel/next.js/pull/51065. I also didn't observe AvaTax suspicious behaviour on PR while testing

Comment (krzysztofzuraw) [reply apps/avatax/tsconfig.json:12]: Also if it helps our payment apps have `bundler` set as well 😄

**Review by lkostrowski — APPROVED** [review]

**Review by lucas-dzi — APPROVED** [review]

**Pull request merged**